### ISSUE TITLE
Custom Notifications

### ISSUE THREAD

**Issue by MadamJazzy** [powercord-community/suggestions#62]
**Describe the plugin**
This plugin would allow users to setup a list of keywords that would be watched for and receive a notification on for all servers/channels they have access to. 
Config options: (examples are in brackets)
- Comma seperated keyword list. [Sapphire, Jazzy, Jazzi] {Case insensative}
- Notify on Mentions [On or off to include mention notifications]
- Notification Type [See below]
- Webhook URL for notifications 
- Message Context(Webhook only) [Numerical input 1-3] {This number represents the amount of messages to obtain before and after the message that contains the keyword. For example if set to 2 it will gather the last 2 messages before the message with the keyword and the 2 messages after the message with the keyword to display a "context" in the webhook embed. 
Notification Types: 
- Via Standard Pop up notification 
- Via Webhook 
Information displayed
- For standard pop up will display "<Username> mentioned <Keyword> in <Server>-<Channel> at <Date/Time>
- For Webhook notification type an embed will be sent to the webhook with the following format
Title: <Username> mentioned <Keyword> in <Server>-<Channel> at <Date/Time>
Description: <Context message> 
{Example of Embed:} https://camo.githubusercontent.com/000ae26f91f1224656d5c8552c498363defbaee5/68747470733a2f2f692e696d6775722e636f6d2f35644d4a6f32372e706e67

**Expected behavior**
Sends a notification when anyone mentions a keyword that is defined in settings to watch for. Will depend on notification type how detailed the notification is. 

----

 - [x] I checked if this request wasn't already proposed
 - [x] I made sure this request follows Powercord's [plugin guidelines](https://github.com/powercord-community/guidelines)

**Comment by CodingKanin** [comment]
I believe this is a duplicate of #1, correct me if I'm wrong, I know your idea has a bit more options but I believe you could comment on that issue with your request.

**Comment by MadamJazzy** [comment]
I mean #1 refers to being pinged when a keyword is mentioned. This refers to more of being notified of the message and possibly "logging" it via webhook to be reviewed later if the users so chooses.

**Comment by CodingKanin** [comment]
> 
> 
> I mean #1 refers to being pinged when a keyword is mentioned. This refers to more of being notified of the message and possibly "logging" it via webhook to be reviewed later if the users so chooses.

Again, they're both very similar. Please comment on that issue with what you would like added to it, being mentioned and being notified are very similar things and it would make more sense to have both of these in one plugin with settings to toggle what you want. Though I'm not sure if the whole logging thing is allowed, we'll see what @Bowser65 says.

**Comment by Bowser65** [comment]
I'd be against the logging part, especially of messages arround it. While they can provide context for quick understanding of what was happening, it falls under guideline 3 as you're logging End User Data without prior consent (especially knowing not all messages that trigger logging are direct mentions).

**Comment by MadamJazzy** [comment]
What if plugin were modified to only allow embed to remain active for a certain amount of time? I mean i know discord does not allow bots to collect user data and store that off of discord, but its always been well established that messages posted within discord from discord don't fall under end user data as it is not leaving the discord application (ie being logged and/or stored away from the discord application). [Not trying to argue just asking :) ]

**Comment by Bowser65** [comment]
Legally-wise, messages are user-generated content that belongs to the User. The User grants a license to Discord for the use of their messages, granting Discord basically all rights on the message sent on the platform (which is required in order to process it). Discord allows bots to use this content and log it as long as the message is logged **in the same guild** (private admin channels for example), or stored in the bot's databsae if the data is required in order to provide the service.

Modifying the plugin to delete it after a period of time isn't a viable solution, as it requries the plugin to be running to actually take effect and delete the message as expected (unless you run an external process but that'd be selfbotting and strictly prohibited).

As we've decided to keep Powercord's ecosystem as clean and awesome as possible, we've decided to be strict on what plugins can do, even if that's restricting features that would be useful, but sadly abusable.
